File: src/main/java/org/gecko/view/views/shortcuts/AutomatonEditorViewShortcutHandler.java
package org.gecko.view.views.shortcuts;

import java.util.List;
import org.gecko.actions.ActionManager;
import org.gecko.tools.ToolType;
import org.gecko.view.views.EditorView;

public class AutomatonEditorViewShortcutHandler extends ShortcutHandler {
    public AutomatonEditorViewShortcutHandler(ActionManager actionManager, EditorView editorView) {
        super(actionManager, editorView);

        addCreatorShortcuts();
    }

    private void addCreatorShortcuts() {
        List<ToolType> creatorTools = List.of(ToolType.STATE_CREATOR, ToolType.EDGE_CREATOR, ToolType.REGION_CREATOR);
        creatorTools.forEach(tool -> {
            shortcuts.put(tool.getKeyCodeCombination(),
                () -> actionManager.run(actionFactory.createSelectToolAction(tool)));
        });
    }
}

File: src/main/kotlin/org/gecko/view/views/shortcuts/AutomatonEditorViewShortcutHandler.kt
package org.gecko.view.views.shortcuts

import org.gecko.actions.ActionManager
import org.gecko.tools.ToolType
import org.gecko.view.views.EditorView
import java.util.List
import java.util.function.Consumer

/**
 * A concrete representation of a [ShortcutHandler] that manages the shortcuts corresponding to the selection of
 * creator tools specific to an automaton editor view.
 */
class AutomatonEditorViewShortcutHandler(actionManager: ActionManager, editorView: EditorView) :
    ShortcutHandler(actionManager, editorView) {
    init {
        addCreatorShortcuts()
    }

    fun addCreatorShortcuts() {
        val creatorTools = List.of(ToolType.STATE_CREATOR, ToolType.EDGE_CREATOR, ToolType.REGION_CREATOR)
        creatorTools.forEach { tool: ToolType ->
            shortcuts[tool.keyCodeCombination] =
                { actionManager.run(actionFactory.createSelectToolAction(tool)) }
        }
    }
}
